go/vt/wrangler/vdiff.go

```diff
@@ -456,6 +460,66 @@ func findPKs(table *tabletmanagerdatapb.TableDefinition, targetSelect *sqlparser
 	return orderby, nil
 }
 
+// If SourceTimeZone is defined in the BinlogSource, the VReplication workflow would have converted the datetime
+// columns expecting the source to have been in the SourceTimeZone and target in TargetTimeZone. We need to do the reverse
+// conversion in VDiff before comparing to the source
+func (df *vdiff) adjustForSourceTimeZone(targetSelectExprs sqlparser.SelectExprs, fields map[string]querypb.Type) sqlparser.SelectExprs {
+	if df.sourceTimeZone == "" {
+		return targetSelectExprs
+	}
+	log.Infof("found source time zone %s", df.sourceTimeZone)
+	var newSelectExprs sqlparser.SelectExprs
+	var modified bool
+	for _, expr := range targetSelectExprs {
+		converted := false
+		switch selExpr := expr.(type) {
+		case *sqlparser.AliasedExpr:
+			if colAs, ok := selExpr.Expr.(*sqlparser.ColName); ok {
+				var convertTZFuncExpr *sqlparser.FuncExpr
+				colName := colAs.Name.Lowered()
+				fieldType := fields[colName]
+				if strings.EqualFold(fieldType.String(), "DATETIME") {
```

Comment:
Curious why we do this instead of looking at the actual type? The UCA string comparison will be much more expensive.